### DISCUSSION TITLE
reCaptcha config problem

### DISCUSSION THREAD

**Discussion by norato777** [emailjs-com/help#71] [Q&A]
When I configure my template to support reCaptcha v2 verification and I enter the secret key and test the template, I get the error "400 reCAPTCHA: The g-recaptcha-response parameter not found". Likewise, from my react app it doesn't work. Can someone tell me what else I should configure or only with the secret key should I do it! thank you

**Comment by xr0master** [comment]
You need to install the captcha itself on your website.
https://developers.google.com/recaptcha/docs/display

**Reply by dickomoa** [reply]
Hi , 
I still have the same issue with mine : 


![image](https://github.com/emailjs-com/help/assets/137198715/d092d569-a1ab-489d-b073-172170746d35)

`const handleSubmit = async (
        values: MyFormValues,
        { setSubmitting, resetForm }: FormikHelpers<MyFormValues>,
    ) => {
        console.time("TraitementServeur");
        const recaptchaToken = recaptchaRef.current ? recaptchaRef.current.getValue() : null;
       
        // Vérifier si le ReCAPTCHA a été coché
        if (!recaptchaToken) {
            // Si le ReCAPTCHA n'est pas coché, afficher un message d'erreur
            setRecaptchaTokenError(true);
            console.log("inside recaptcha not working");
            //toast.error('Veuillez cocher la case ReCAPTCHA.');
            setSubmitting(false);
            return;
        }
        try {
         
            const emailjsServiceId = import.meta.env.VITE_EMAILJS_SERVICE_ID;
            const emailjsTemplateId = import.meta.env.VITE_EMAILJS_TEMPLATE_ID;
            const emailjsUserId = import.meta.env.VITE_EMAILJS_USER_ID;
            const templateParams = {
              prenom: values.prenom,
              nom: values.nom,
              telephone: values.telephone,
              email: values.email,
              societe: values.societe,
              message: values.message,
            };
            const emailJsResult = await emailjs.send(
              emailjsServiceId,
              emailjsTemplateId,
              templateParams,
              emailjsUserId
            );

            if (emailJsResult.status === 200) {
                setRecaptchaTokenError(false);
                toast.success("Formulaire soumis avec succès et email envoyé!");
                console.timeEnd("TraitementServeur"); // Arrêter le chronomètre après l'affichage de la notification
            } else {
                console.timeEnd("TraitementServeur");
                toast.error("Erreur lors de la soumission du formulaire.");
            }
        } catch (error) {
            toast.error("Erreur lors de la soumission du formulaire.");
            console.timeEnd("TraitementServeur");
        }
        recaptchaRef.current ? recaptchaRef.current.reset() : null;
        setSubmitting(false);
        resetForm();
    };
return ( 
  ...............
  <div className={recaptchaTokenError ? "recaptcha-wrapper" : ""}>
                            <ReCAPTCHA
                                ref={recaptchaRef}
                                sitekey={import.meta.env.VITE_APP_GOOGLE_RECAPTCHA_SITE_KEY}
                                //onChange={handleRecaptchaChange }
                                size="normal"
                            />
                        </div>
)`

**Reply by xr0master** [reply]
Hi, I don't know what is the error message, but most likely you need to add `g-recaptcha-response` variable.
```js
const templateParams = {
  prenom: values.prenom,
  nom: values.nom,
  telephone: values.telephone,
  email: values.email,
  societe: values.societe,
  message: values.message,
  'g-recaptcha-response': recaptchaToken,
};
```

Please note the `g-recaptcha-response` is how Google called this variable.

**Reply by dickomoa** [reply]
Hi , 
Thank you for the response , but still not working
